src/components/TicketModal/TicketModal.vue

```diff
@@ -2,175 +2,268 @@
+        selectedUser.value.username = "None";
+        selectedMember.value.id = "-1";
+      } catch (error) {
+        alert(error);
+      }
     };
-    const closeModal = () => {
-      context.emit("close");
+    const getStatuses = async () => {
+      try {
+        statuses.value = Object.values(
+          await listStatuses(selectedTeam.value.id)
+        );
+        selectedStatus.value.title = statuses.value[0].title;
+        selectedStatus.value.color = statuses.value[0].color;
+        selectedStatus.value.id = statuses.value[0].id;
+      } catch (error) {
+        alert(error);
+      }
+    };
+    const statusSelected = (display: string, item: StatusRecordOutput) => {
+      selectedStatus.value.title = display;
+      selectedStatus.value.color = item.color;
+      selectedStatus.value.id = item.id;
+    };
+    const userSelected = (display: string, item: MemberRecord) => {
+      selectedUser.value.username = display;
+      selectedUser.value.pk = item.id;
+      selectedMember.value.id = item.id;
+    };
+    const teamSelected = async (display: string, item: ReadTeamRecord) => {
+      if (selectedTeam.value.id !== item.id) {
+        selectedTeam.value.name = display;
+        selectedTeam.value.id = item.id;
+        await getTags();
+        await getMembers();
+        await getStatuses();
+      }
+    };
+    const tagSelected = (display: string, item: TagRecord) => {
+      selectedTag.value.title = display;
+      selectedTagId.value[0] = item.id;
+    };
+    const setTitle = (item: string) => {
+      ticketObj.value.title = item;
+      shouldShowPencil.value = true;
     };
-    const saveChanges = () => {
+    const closeModal = () => {
       context.emit("close");
     };
-    const setTicketData = () => {
-      if (props.ticketRecord) {
-        const ticket = props.ticketRecord;
-        ticketStatus.value = ticket.status?.title || "";
-        ticketUser.value = ticket.assigned_user?.username || "";
+    const saveChanges = async () => {
+      const ticket: WriteTicketRecord = {
+        title: ticketObj.value.title,
+        status: selectedStatus.value.id
+      };
+      if (ticketObj.value.due_date) {
+        ticket.due_date = ticketObj.value.due_date.replace(/T/, " ");
+      }
+      if (selectedMember.value.id !== "-1") {
+        ticket.assigned_user = selectedMember.value.id;
+      }
+      if (ticketObj.value.description !== "") {
+        ticket.description = ticketObj.value.description;
+      }
+      if (selectedTagId.value[0] !== -1) {
+        ticket.tag_list = selectedTagId.value;
+      }
+      try {
+        await createTicket(ticket, selectedTeam.value.id);
+      } catch (error) {
+        alert(error);
       }
+      closeModal();
     };
-    onMounted(setTicketData);
+    onMounted(async () => {
+      initializeData();
+      await getTeams();
+      await getTags();
+      await getMembers();
+      await getStatuses();
```

Comment:
Same thing here. If the requests don't depend on each other, we might find a performance boost by using `Promise.all`